pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -243,6 +258,33 @@ func NewController(kubeClient kubelib.Client, options Options) *Controller {
 	c.serviceLister = kubeClient.KubeInformer().Core().V1().Services().Lister()
 	registerHandlers(c.serviceInformer, c.queue, "Services", c.onServiceEvent, nil)
 
+	if enableEventing.Get() {
+		// To get events posted by other Istiod instances, in order to sync and support debug tools.
+		// This will also allow us to get readiness/status when VM health check is done, and possibly get faster
+		// updates about endpoints.
+		c.eventsInformer = kubeClient.KubeInformer().Core().V1().Events().Informer()
+		c.eventsInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				if ev, ok := obj.(*corev1.Event); ok {
+					c.onEvent(ev, "add")
+				}
+			},
+			UpdateFunc: func(old, obj interface{}) {
+				if ev, ok := obj.(*corev1.Event); ok {
+					c.onEvent(ev, "update")
```

Comment:
So far it doesn't seem possible to filter by type of event - and since we just drop the events we don't care about it doesn't seem to create any major problem. 

We do have a flag to control the feature, and we will have additional options - like using NATS/etc  for even higher event volume ( real pubsub systems allow filtering ). 

In general the Event volume is about proportional with the pod events - the rest are far less volume. And we need to scale linearly with pod events - since we watch them. Of course load testing and benchmarks will be used before this feature is released.